src/benchmark_register.cc

```diff
@@ -461,13 +490,50 @@ int Benchmark::ArgsCnt() const {
   return static_cast<int>(args_.front().size());
 }
 
-//=============================================================================//
-//                            FunctionBenchmark
-//=============================================================================//
+void Benchmark::ApplyFunctions() {
+  if (functions_to_apply_.size()) {
+    for (auto f: functions_to_apply_) {
+      f(this);
+    }
+    functions_to_apply_.clear();
+    functions_to_apply_.shrink_to_fit();
+  }
+}
 
 void FunctionBenchmark::Run(State& st) { func_(st); }
 
-}  // end namespace internal
+void FixtureBenchmark::Run(State& st) {
+  fixture_->SetUp(st);
+  fixture_->BenchmarkCase(st);
+  fixture_->TearDown(st);
+}
+
+void FixtureBenchmark::TryCreateFixture() {
+  if (!fixture_) {
+    fixture_.reset(creator_());
+  }
+}
+
+Benchmark* RegisterBenchmarkInternal(Benchmark* p_benchmark) {
+  BenchmarkFamilies::GetInstance()->AddBenchmark(std::unique_ptr<Benchmark>(p_benchmark));
+  return p_benchmark;
+}
+
+} // end namespace internal
+
+// register a function benchmark
```

Comment:
this documentation doesn't add anything.

Reply:
I will change the code.